### ISSUE TITLE
real loop indexes used in mbpf

### ISSUE THREAD

**Issue by dkokron** [NOAA-EMC/GSI#757]
GDIT is testing a static analysis tool called Codee (https://www.codee.com/).  Codee flagged some issues with mgbf.

codee: error: failed to analyze '/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90'

error: Semantic errors in /lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:67:12: error: Must have INTEGER type, but is REAL(4)
           V(n,m)=ampl*(mg*imax0+ng)
             ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:67:14: error: Must have INTEGER type, but is REAL(4)
           V(n,m)=ampl*(mg*imax0+ng)
               ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:145:12: error: Must have INTEGER type, but is REAL(4)
           V(l,n,m)=ampl*(mg*imax0+ng) +(l-1)*incrm
             ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:145:14: error: Must have INTEGER type, but is REAL(4)
           V(l,n,m)=ampl*(mg*imax0+ng) +(l-1)*incrm
               ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:145:16: error: Must have INTEGER type, but is REAL(4)
           V(l,n,m)=ampl*(mg*imax0+ng) +(l-1)*incrm

The l,n,m variables are loop indexes but they are declared.
real(i_kind):: ng,mg,L,m,n

**Comment by TingLei-NOAA** [comment]
@dkokron  Thanks for this validation effort.    The apparent fix is to use " integer(i_kind)" to replace "real(i_kind)". 
I think you can direct make  a PR for this or I can do it if you prefer. 
Please let me know how you want to proceed. Thanks. 
BTW: it is mgbf rather than "mbpf".

**Comment by dkokron** [comment]
@TingLei-NOAA Please submit the PR for this issue.

**Comment by dkokron** [comment]
I thought Ting was handling this issue.

**Comment by TingLei-NOAA** [comment]
@dkokron  You identified the problem and created the issue and you are also the coauthor for that PR.  Finally, if you are satisfied the solution when the PR is finally merged into the develop branch, you could close this issue as the assignee.

**Comment by TingLei-daprediction** [comment]
An update:  the PR with the quick fix for the issue had been closed and will be re-opened pending on future mgbf upgrading (https://github.com/NOAA-EMC/GSI/pull/760). 
It is also noted the program causing the safety concern identified in this would only be used when the mgbf package is used as the stand-alone version using its own driver and initialization blocks(which would invoke this code).

**Comment by RussTreadon-NOAA** [comment]
PR #760 was closed because there was no timeline for working on the PR.  This issue will remain open until the bug is fixed.  Clearly the [declaration](https://github.com/NOAA-EMC/GSI/pull/760/files) of `integer(i_kind)` variables as `real(i_kind)` in `src/mgbf/mg_input.f90` is wrong.
